plaso/parsers/onedrive.py

```diff
@@ -0,0 +1,449 @@
+# -*- coding: utf-8 -*-
+"""Parser for OneDrive Log (ODL/ODLGZ) files.
+
+Reference: https://forensics.wiki/microsoft_onedrive/
+"""
+
+import base64
+import io
+import json
+import os
+import re
+import zlib
+
+import pycaes
+
+from cryptography.hazmat.primitives import padding
+
+from dfdatetime import posix_time
+
+from dfvfs.helpers import text_file
+from dfvfs.resolver import resolver as path_spec_resolver
+from dfvfs.path import factory as path_spec_factory
+
+from plaso.containers import events
+from plaso.lib import dtfabric_helper
+from plaso.lib import errors
+from plaso.lib import specification
+from plaso.parsers import interface
+from plaso.parsers import manager
+
+
+class OneDriveLogEvent(events.EventData):
+  """OneDrive log event data.
+
+  Attributes:
+    code_file_name (str): code file name
+    code_function_name (str): code function name
+    decoded_parameters (str): decoded (and decrypted) parameters
+    raw_parameters (str): the raw parameters encoded as a hex string
+    recorded_time (dfdatetime.DateTimeValues): date and time the entry was
+        recorded.
+  """
+
+  DATA_TYPE = 'windows:onedrive:log'
+
+  def __init__(self):
+    """Initializes event data."""
+    super(OneDriveLogEvent, self).__init__(data_type=self.DATA_TYPE)
+    self.code_file_name = None
+    self.code_function_name = None
+    self.decoded_parameters = None
+    self.raw_parameters = None
+    self.recorded_time = None
+
+
+class OneDriveLogFileParser(
+    interface.FileEntryParser, dtfabric_helper.DtFabricHelper):
+  """Parser for OneDrive log files."""
+
+  NAME = 'onedrive_log'
+  DATA_FORMAT = 'OneDrive Log file'
+
+  BLOCK_SIGNATURE = b'\xcc\xdd\xee\xff\x00\x00\x00\x00'
+  COMPRESSED_BLOCK_SIGNATURE = b'\x1f\x8b\x08\x00\x00\x00\x00\x00'
+  FORMAT_SIGNATURE = b'EBFGONED'
+
+  _AES_IV = b'\x00' * 16
+  _BASE64_ALPHABET = (
+      'ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789_-')
+  _DEFINITION_FILE = os.path.join(
+      os.path.dirname(__file__), 'onedrive.yaml')
+  _DELIMITERS = [r'%', r'\\', r'/', r'://', r'\?', r':', r'\.']
+  _DELIMITERS_REGEX = '({0:s})'.format('|'.join(_DELIMITERS))
+  _GENERAL_KEYSTORE_FILE = 'general.keystore'
+  _OBFUSCATED_STRING_MAP_FILE = 'ObfuscationStringMap.txt'
+
+  @classmethod
+  def GetFormatSpecification(cls):
+    """Retrieves the format specification.
+
+    Returns:
+      FormatSpecification: format specification.
+    """
+    format_specification = specification.FormatSpecification(cls.NAME)
+    format_specification.AddNewSignature(cls.FORMAT_SIGNATURE, offset=0)
+    return format_specification
+
+  def _OpenFileInSameDirectory(
```

Comment:
Technically you're not "opening"  a file here but retrieving its file (system) entry

Reply:
Updated.